opentelemetry-api/src/opentelemetry/correlationcontext/__init__.py

```diff
@@ -33,8 +34,8 @@ def get_correlations(
     """
     correlations = get_value(_CORRELATION_CONTEXT_KEY, context=context)
     if isinstance(correlations, dict):
-        return correlations.copy()
-    return {}
+        return MappingProxyType(correlations.copy())
```

Comment:
Do we still need to copy the context if we are returning a read-only mapping?

Reply:
The returned proxy view is read-only, but the underlying object is not. Creating a new copy ensures that there is no mutable reference to it around elsewhere.

Reply:
But how could you access the underlying object outside of `get_correlations`?

Reply:
The object is a member of the `Context` object that is an optional argument to the function. So presumably every such caller, or anyone else with access to the context, could just access the context dict, and therefore the correlations, directly.

Reply:
I am more asking whether if returning a stale copy of what is in the correlation context is actually what we want? Like you mentioned, it is possible for this member in the `Context` dict to change, do we want `get_correlations` to reflect these changes? If that's the case, we don't want to be creating as copy of it.

Reply:
I was under the impression that immutability refers to the object’s state, not just its API. Returning a stale copy is also how it was implemented previously, so I honestly did not even consider that we might want to reflect any changes.

However, the spec says that `get_correlations` returns the name/value pairs in the `CorrelationContext`, rather than the context itself, so I don’t think an up-to-date view on the object is what is intended. The correlation modification operations also all create a new `Context`, and while not explicitly stated, it reads to me like this context also contains a new `CorrelationContext`, meaning that these changes would not be reflected in the object returned by `get_correlations` either way.

Reply:
Yeah, reading it now that seems to be my understanding as well.